frontend/src/container/Favorites.jsx

```diff
@@ -0,0 +1,292 @@
+      </div>
+      {/* Header for windows smaller than max-sm */}
+      <div className='favoritesHeader__title-max-sm'>
+        {props.t('Title')}
+      </div>
+      <div className='favoritesHeader__modification'>
+        {props.t('Last Modification')}
+      </div>
+      <div className='favoritesHeader__information'>
+        {props.t('Information_plural')}
+      </div>
+      <div className='favoritesHeader__favoriteButton'>
+        {props.t('Favorite')}
+      </div>
+    </div>
+  )
+})
+
+const UnavailableContent = translate()(props => {
+  return (
+    <ListItemWrapper
+      label={props.label}
+      read
+      contentType={props.contentTypeInfo}
+      isLast={props.isLast}
+      customClass='unavailableContent contentListItem'
+    >
+      <ContentType
+        contentTypeInfo={props.contentTypeInfo}
+        customClass='contentListItem__type'
+      />
+      <div className='contentListItem__name_path unavailableContent__name_warning'>
+        {props.label}
+        <span className='unavailableContent__warning'>
+          <Icon
+            icon='fa-fw fas fa-exclamation-triangle'
+            title={props.t('Warning')}
+          />
+          &nbsp;
+          {props.t('content is not available')}
+        </span>
+      </div>
+      {props.children}
+    </ListItemWrapper>
+  )
+})
+
+export class Favorites extends React.Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      contentCommentsCountList: [],
+      contentBreadcrumbsList: []
+    }
+
+    props.registerCustomEventHandlerList([
+      { name: CUSTOM_EVENT.ALL_APP_CHANGE_LANGUAGE, handler: this.handleAllAppChangeLanguage }
+    ])
+  }
+
+  handleAllAppChangeLanguage = () => {
+    this.setHeadTitle()
+    this.buildBreadcrumbs()
+  }
+
+  setHeadTitle = () => {
+    const { props } = this
+    const headTitle = buildHeadTitle([props.t('Favorites')])
+    props.dispatch(setHeadTitle(headTitle))
+  }
+
+  buildBreadcrumbs = () => {
+    const { props } = this
+
+    props.dispatch(setBreadcrumbs([{
+      link: PAGE.FAVORITES,
+      type: BREADCRUMBS_TYPE.CORE,
+      label: props.t('Favorites'),
+      isALink: true
+    }]))
+  }
+
+  componentDidMount () {
+    this.setHeadTitle()
+    this.buildBreadcrumbs()
+    this.loadFavoriteList()
+  }
+
+  loadFavoriteList = async () => {
+    const { props } = this
+
+    const fetchFavoriteList = await props.dispatch(
+      getFavoriteContentList(props.user.userId)
+    )
+
+    if (!fetchFavoriteList.ok) {
+      props.dispatch(newFlashMessage(props.t('An error has happened while fetching favorites'), 'warning'))
+      return
+    }
+    const favoriteList = fetchFavoriteList.json.items
+    // Get comments (for their count in info)
+    const commentsFetchList = favoriteList.map(async favorite => {
+      if (!favorite.content) return null
+      // NOTE - S.G. - 2021-04-01 - here we have the favorite as returned by the backend
+      // hence the snake-case properties
```

Comment:
All this object serialization/where it should be done is quite unclear in my eyes. I saw that most of these serialize is done in the reducers so I did it likewise (as you'll find in `favoriteList.js`), but here this code is executed before the reducer.

Reply:
A reminder that serialization needs to be addressed once and for all at some point: #3742

Reply:
the point of the serialize functions is to avoid manipulating snake case variables
It is in the reducer because it usually is the first thing called after an api call but it doesn't mean you should only use it here
Feel free to use it everywhere you need to manipulate an api object

Reply:
OK, but if i use it here the reducer won't perform it, unlike many others… which could be surprising later on.